plugins/permission-common/src/types/permission.ts

```diff
@@ -39,3 +41,18 @@ export type Permission = {
   attributes: PermissionAttributes;
   resourceType?: string;
 };
+
+export interface PermissionClientInterface {
+  authorize(
+    requests: AuthorizeRequest[],
+    options?: AuthorizeRequestOptions,
+  ): Promise<AuthorizeResponse[]>;
+}
+
+/**
+ * Options for authorization requests; currently only an optional auth token.
```

Comment:
```suggestion
 * Options for authorization requests.
```

the other contextual info can go on the field itself